### PR TITLE
Add support for hiding posts

### PR DESCRIPTION
**Pull request by micahmo**
## Pull Request Description

This PR adds support for the ability to hide posts introduced in Lemmy 0.19.4.
- There is a new swipe action and a new post sheet action.
- The action color can be customized.
- There is a new icon on posts that indicates whether they are hidden or not.

Unfortunately there does not seem to be a Lemmy account setting for whether to show/hide hidden posts in the feed, so for now I've added a Thunder setting. But we can migrate this to an account setting if it ever becomes available.

Notes:
* If you configure the hide gesture and then use an instance which does not support the feature, the gesture will not be available. The post sheet action will also be hidden.
* Unlike the web UI, hiding a post from Thunder will immediately dismiss it from the feed (if the user has chosen not to display hidden posts).
  * Auto-dismissal from the user's own user page is not yet supported, since it does not yet use the same feed widget as the general feed and other user pages. However, the hiding feature still works.

## Issue Being Fixed

Issue Number: #1447

## Screenshots / Recordings

Demo 1: Hide via swipe and button.

https://github.com/thunder-app/thunder/assets/7417301/5d3c1123-f486-4a46-b645-65aa14a58f0e

Demo 2: Hide via swipe and button, respecting show hidden. Then unhide.

https://github.com/thunder-app/thunder/assets/7417301/57ffcca0-65a6-4a35-a77e-4304205b5466

Demo 3: Swipe and button are not visible on older instances.

https://github.com/thunder-app/thunder/assets/7417301/69ee5f57-e100-46d1-a219-b553675178c0

## Checklist

- [ ] If a new package was added, did you ensure it uses an appropriate license and is actively maintained?
- [ ] Did you use localized strings (and added appropriate descriptions) where applicable?
- [ ] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

File: lib/community/utils/post_card_action_helpers.dart [hunk 1]
```diff
@@ -243,6 +244,15 @@ final List<ExtendedPostCardActions> postCardActionItems = [
     getOverrideIcon: (postView) => postView.read ? Icons.mark_email_unread_rounded : Icons.mark_email_read_outlined,
     shouldEnable: (isUserLoggedIn) => isUserLoggedIn,
   ),
+  ExtendedPostCardActions(
+    postCardAction: PostCardAction.hide,
+    label: l10n.hide,
+    getOverrideLabel: (context, postView) => postView.hidden == true ? l10n.unhide : l10n.hide,
+    icon: Icons.close_rounded,
```
Comment (hjiangsu) [inline lib/community/utils/post_card_action_helpers.dart:251]: Thoughts about using `visibility` and `visibility_off` for the icons?

File: lib/community/widgets/post_card_view_comfortable.dart [hunk 2]
```diff
@@ -113,6 +114,15 @@ class PostCardViewComfortable extends StatelessWidget {
               child: Text.rich(
                 TextSpan(
                   children: [
+                    if (postViewMedia.postView.hidden == true)
+                      WidgetSpan(
+                        child: Icon(
+                          Icons.close_rounded,
```
Comment (hjiangsu) [inline lib/community/widgets/post_card_view_comfortable.dart:120]: If we're changing the icon for the action, this should also be updated to reflect that! (and any other instances)

**Review by hjiangsu — APPROVED** [review]
Just a minor comment! Otherwise LGTM

File: lib/community/utils/post_card_action_helpers.dart (hunk 1 above)
Comment (micahmo) [reply lib/community/utils/post_card_action_helpers.dart]: Ah yes those match the Lemmy UI. Perfect. Should be good to go now!

**Pull request merged**